Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -419,22 +419,22 @@ public class BrowserViewController: UIViewController {
     updateApplicationShortcuts()
     tabManager.addDelegate(self)
     tabManager.addNavigationDelegate(self)
-    tabManager.makeWalletEthProvider = { [weak self] tab in
-      guard let self = self,
-            let provider = self.braveCore.braveWalletAPI.ethereumProvider(with: tab, isPrivateBrowsing: tab.isPrivate),
-            let js = self.braveCore.braveWalletAPI.providerScripts(for: .eth)[.ethereum] else {
-        return nil
-      }
-      return (provider, js: js)
-    }
-    tabManager.makeWalletSolProvider = { [weak self] tab in
-      guard let self = self,
-            let provider = self.braveCore.braveWalletAPI.solanaProvider(with: tab, isPrivateBrowsing: tab.isPrivate) else {
-        return nil
-      }
-      let scripts = self.braveCore.braveWalletAPI.providerScripts(for: .sol)
-      return (provider, jsScripts: scripts)
+    if let ethJS = braveCore.braveWalletAPI.providerScripts(for: .eth)[.ethereum] {
+      let providerJS = """
+      window.__firefox__.execute(function($, $Object) {
+        if (window.isSecureContext) {
+          \(ethJS)
+        }
+      });
+      """
+      UserScriptManager.shared.walletEthProviderScript = WKUserScript(
+        source: providerJS,
+        injectionTime: .atDocumentStart,
+        forMainFrameOnly: true,
+        in: EthereumProviderScriptHandler.scriptSandbox
+      )
     }
+    UserScriptManager.shared.walletSolProviderScripts = braveCore.braveWalletAPI.providerScripts(for: .sol)
```

Comment:
Lets move this logic into `UserScriptManager`, maybe a function that just takes a BraveWalletAPI, sets up all wallet providers inside of it, that way we can make these `WKUserScripts` private like the rest and that custom JS also sticks with the rest of the scripts

Reply:
Thanks, updated in [66a03b8](https://github.com/brave/brave-ios/pull/7483/commits/66a03b8a0a8aa8c07905dce97e8bbfe42f57a1d6) cc @Brandon-T